policy/executor/internal/collector.go

```diff
@@ -86,12 +86,12 @@ func (c *BufferedCollector) run() {
 			}
 			c.lock.Unlock()
 
-			if len(results) > 0 {
-				c.collector.SinkData(results)
+			if len(results) > 0 || done {
```

Comment:
More for my understanding of this logic:
This will send an emptyl StoreResults to the server after we saved all the items?

Reply:
yes, we always need to send a `StoreResults` with `IsLastBatch = true`. If it happens to be that we have no data to send when we called `FlushAndStop` we should still make the call to notify the server we are done